database/meta.go

```diff
@@ -86,9 +86,20 @@ func (m *MetaDBHandle) DeleteDB(name string) error {
 	ssn.DB(name).DropDatabase()
 
 	//delete any parsed file records associated
-	_, err = ssn.DB(m.DB).C("files").RemoveAll(bson.M{"database": name})
-	if err != nil {
-		return err
+	if db.UsingDates {
```

Comment:
We need to remove the file entries which match the database to be removed. We split the database name and remove any files which were destined for the base database name and contained entries corresponding to the date.